providers/aws-ec2/src/test/java/org/jclouds/aws/ec2/features/RouteTableApiLiveTest.java

```diff
@@ -0,0 +1,242 @@
+            assertEquals(defaultCidr, vpc.cidrBlock(),
+               "Route in default route table does not match CIDR of VPC, " + defaultCidr + " should be " + vpc.cidrBlock());
+            break;
+         }
+      }
+      assertTrue(foundVpc, "Could not find VPC " + vpc.id() + " in described route tables");
+   }
+
+   @Test(dependsOnMethods = "testDescribe")
+   public void testCreate() {
+
+      // When you create a new route table for the VPC it automatically gets a route to match the VPC CIDR
+      routeTable = routeTableApi.createRouteTable(TEST_REGION, vpc.id());
+      assertNotNull(routeTable, "Gateway was not successfully created");
+
+      assertEquals(routeTable.vpcId(), vpc.id(),
+         "RouteTable VPC ID " + routeTable.vpcId() + " does not match VPC's ID " + vpc.id());
+      final List<Route> routes = routeTable.routeSet();
+      assertEquals(routes.size(), 1, "Unexpected number of routes in new table: " + routes.size());
+      assertEquals(routes.get(0).destinationCidrBlock(), vpc.cidrBlock(),
+         "CIDR for route table " + routes.get(0).destinationCidrBlock() +
+            " does not match VPC CIDR" + vpc.cidrBlock());
+      assertEquals(routes.get(0).state(), Route.RouteState.ACTIVE, "Route should be active");
+      assertEquals(routeTable.tags().size(), 0, "Freshly created routeTable has tags");
+
+      tagger.applyToResources(ImmutableMap.of("Name", simpleName), ImmutableList.of(routeTable.id()));
+      getAnonymousLogger().info("Created routeTable " +  simpleName + " with id " + routeTable.id());
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testAssociate() {
+      subnet = subnetApi.createSubnetInRegion(TEST_REGION, vpc.id(), VPC_SUBNET);
+      assertNotNull(subnet, "Failed to create subnet in " + vpc.id());
+
+      associationId = routeTableApi.associateRouteTable(TEST_REGION, routeTable.id(), subnet.getSubnetId());
+      assertNotNull(associationId,
+         "Failed to obtain association id for " + routeTable.id() + " and " + subnet.getSubnetId());
+
+      routeTable = routeTableApi.describeRouteTables(TEST_REGION, routeTable.id()).toList().get(0);
+      assertEquals(routeTable.associationSet().size(), 1,
+         "Could not find expected association in routeTable " + routeTable.id());
+   }
+
+   @Test(dependsOnMethods = "testAssociate")
+   public void testDisassociate() {
+      final boolean result = routeTableApi.disassociateRouteTable(TEST_REGION, associationId);
+      assertTrue(result, "Failed to disassociate " + associationId + " from " + routeTable.id());
+
+      routeTable = routeTableApi.describeRouteTables(TEST_REGION, routeTable.id()).toList().get(0);
+      assertEquals(routeTable.associationSet().size(), 0,
+         "Found associations where none should exist in  " + routeTable.id() + ": " + routeTable.associationSet());
+
+      subnetApi.deleteSubnetInRegion(TEST_REGION, subnet.getSubnetId());
+   }
+
+   @Test(dependsOnMethods = "testDisassociate")
```

Comment:
Can it just depend on the `testCreate` to maximize the chances of this test being run in case the associate tests failed?